### DISCUSSION TITLE
Wrong Imported Time

### DISCUSSION THREAD

**Discussion by recke96** [bitfireAT/icsx5#188] [Q&A] [answered]
I'm trying to subscribe to the calendar at <https://www.marozzo.info/event?ical>.

Unfortunately, the time of the events is off, and not by a consistent duration (18:45 → 01:00, 18:15 → 01:00).
The calendar works with GNOME Calendar.

<details>

<summary>Example pictures</summary>

First example showing error in ICSx5
![First example showing error in ICSx5](https://github.com/bitfireAT/icsx5/assets/7427710/fd2699c1-c591-4198-a303-07aa6bd27c22)

Second example showing error in ICSx5
![Second example showing error in ICSx5](https://github.com/bitfireAT/icsx5/assets/7427710/25dece71-ce3f-4a82-92cd-a152f50694fb)

First working example
![First working example](https://github.com/bitfireAT/icsx5/assets/7427710/f1e5b107-d7ed-449a-a55e-3fdb6650fa19)

econd working example
![Second working example](https://github.com/bitfireAT/icsx5/assets/7427710/5ec247f2-3bf9-4b8d-b18a-e97cfb04666e)

</details>

**Comment by tiggle-dd** [comment]
I have the same problem with another ical-feed. Same problem when subscribing to the same feed via DAVx5 — I have subscribed to this feed in my cloud service, there all times are correct.

**Reply by bemmim** [reply]
I have the same problem when subscribing to https://www.criticalmass-muenchen.de/events.ics - what could be the reason? iCalendar validator [only warns about wrong MIME type](https://icalendar.org/validator.html?url=https://www.criticalmass-muenchen.de/events.ics).

**Comment by sunkup** [comment]
Thanks for reporting. We will look into this asap.

**Comment by ArnyminerZ** [comment]
It looks like the problem has been fixed, at least for me. Please, reopen if necessary.

**Reply by sunkup** [reply]
Seems like this still is an issue. I unmarked the answer for now and will try to look into it.

**Comment by lambdadog** [comment]
Hi, I'm having what seems to be this same problem.

Every event in this ICS file shows as being scheduled at 11PM the day before the event happens, regardless of the time of the event, and looking at the ICS file, `DTSTART` and `DTEND` for the events both seem completely correct.

|Event|Expected|Actual|
|-:|:-|:-|
|Alright \| Sinai Vessel \| Flannelmouth|September 8 @ 7:00 pm|September 7 @ 11:00 pm|
|The Bill Hanna Legacy Jazz Session| September 9 @ 7:00 pm|September 8 @ 11:00 pm|
|Karaoke with Klaytastic|September 11 @ 7:00 pm|September 10 @ 11:00 pm|
|Carolina Down Boys \| Paisley Fields \| Coughing Dove|September 12 @ 8:00 pm|September 11 @ 11:00 pm|
|Freaky Friday w/ Jameilyara, Florecienta, & DJ Black As The Cosmos|September 13 @ 8:00 pm|September 12 @ 11:00 pm|
|Swansgate \| Dhemo \| Tecoby Hines|September 14 @ 8:00 pm|September 13 @ 11:00 pm|
|Hazy Sunday|September 15 @ 4:00 pm - 9:00 pm|September 14 @ 11:00 pm|
|...|...|...|
|M143 \| Bonner Black \| Leah Simone|October 11 @ 8:00 pm|October 10 @ 11:00 pm|

ICS file for reference: [petras-bar-33126b5ad11.ics.txt](https://github.com/user-attachments/files/16923421/petras-bar-33126b5ad11.ics.txt) (GitHub does not allow uploads of .ics extension, so uploaded as .txt)

I'm using ICSx<sup>5</sup> version [v2.2.4, from the GitHub Releases page](https://github.com/bitfireAT/icsx5/releases/tag/v2.2.4), on Android version 14. My system timezone is set to `New York (GMT-04:00)` (same as the TZID param for the DTSTART and DTEND entries in the file, if it matters).

**Reply by lambdadog** [reply]
@ArnyminerZ It doesn't seem like I can reopen this issue myself, would you be able to reopen it?

**Comment by q-wertz** [comment]
Having the same issue with https://dokus.uni-stuttgart.de/events.ics
Calendar entries are correct according to https://icalendar.org/validator.html?url=https://dokus.uni-stuttgart.de/events.ics but are shown at 1 o'clock by ICSx5 on Android.

On GNOME Evolution and Outlook the entries are correct (starting at 16:30).

With a different Wordpress plugin somebody described a similar behavior: https://wordpress.org/support/topic/ical-includes-faulty-timezone-definition/

**Comment by sunkup** [comment, marked as answer]
Thank you everyone for providing extra information. I can confirm the events are being imported wrongly. We will look into this again. 

Issue: https://github.com/bitfireAT/icsx5/issues/419
